### PR TITLE
[#18637][TranslationDebug] workaround for getFallbackLocales.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #18637
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php [hunk 1]
```diff
@@ -157,21 +159,30 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-                    continue;
-                }
 
-                $fallbackCatalogue = new MessageCatalogue($fallbackLocale);
-                foreach ($transPaths as $path) {
-                    $path = $path.'translations';
-                    if (is_dir($path)) {
-                        $loader->loadMessages($path, $fallbackCatalogue);
-                    }
+        $fallbackLocales = [];
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php]: `array()`

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php [hunk 2]
```diff
@@ -157,21 +159,30 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        if ($translator instanceof DataCollectorTranslator || $translator instanceof LoggingTranslator) {
+            try {
+                $fallbackLocales = $translator->getFallbackLocales();
+            } catch (\Throwable $e) {
+            }
+        } elseif ($translator instanceof Translator) {
+            $fallbackLocales = $translator->getFallbackLocales();
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php]: What about a new interface for this? Or at least actually define this method? Imo. we should not rely on the magic `__call` behavior of these translators.

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php]: > What about a new interface for this?

seems better!

> Imo. we should not rely on the magic __call behavior of these translators.

right, It was added for BC and I think we should deprecate this behavior

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php]: I defined this method for the moment

**Comment by Simperfit** [comment]
@aitboudad since you introduced two new public method, could you please add tests for them :p ?

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request merged**